Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -266,7 +271,211 @@ public override void DisconnectALMServer()
+            if (bizFlow.ExternalID == "0" || String.IsNullOrEmpty(bizFlow.ExternalID))
+            {
+                result = GingerDicser.GetTermResValue(eTermResKey.BusinessFlow) + ": " + bizFlow.Name + " is missing ExternalID, cannot locate QC TestSet without External ID";
+                return false;
+            }
+            try
+            {
+                //get the BF matching test set
+                QCTestSet testSet = this.GetTestSuiteById(bizFlow.ExternalID);//bf.externalID holds the TestSet TSTests collection id
+                if (testSet != null)
+                {
+                    //get the Test set TC's
+                    QCTestInstanceColl qcTSTests = this.GetTestsFromTSId(testSet.Id); //list of TSTest's on main TestSet in TestLab 
+
+                    //get all BF Activities groups
+                    ObservableList<ActivitiesGroup> activGroups = bizFlow.ActivitiesGroups;
+                    if (activGroups.Count > 0)
+                    {
+                        RunSuite runSuite = CreateRunSuite(publishToALMConfig, bizFlow, testSet, runFields);
+                        foreach (ActivitiesGroup activGroup in activGroups)
+                        {
+                            if ((publishToALMConfig.FilterStatus == FilterByStatus.OnlyPassed && activGroup.RunStatus == eActivitiesGroupRunStatus.Passed)
+                            || (publishToALMConfig.FilterStatus == FilterByStatus.OnlyFailed && activGroup.RunStatus == eActivitiesGroupRunStatus.Failed)
+                            || publishToALMConfig.FilterStatus == FilterByStatus.All)
+                            {
+                                QCTestInstance tsTest = null;
+                                //go by TC ID = TC Instances ID
+                                tsTest = qcTSTests.Find(x => x.TestId == activGroup.ExternalID && x.Id == activGroup.ExternalID2);
+
+                                if (tsTest == null)
+                                {
+                                    //go by Linked TC ID + TC Instances ID
+                                    tsTest = qcTSTests.Find(x => x.Id == activGroup.ExternalID && x.TestId == activGroup.ExternalID2);
+                                }
+                                if (tsTest == null)
+                                {
+                                    //go by TC ID 
+                                    tsTest = qcTSTests.Find(x => x.TestId == activGroup.ExternalID);
+                                }
+                                if (tsTest != null)
+                                {
+                                    //get activities in group
+                                    List<Activity> activities = (bizFlow.Activities.Where(x => x.ActivitiesGroupID == activGroup.Name)).Select(a => a).ToList();
+                                    //Changed this because tsTest.Name is null. string TestCaseName = PathHelper.CleanInValidPathChars(tsTest.Name);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6361231)